### PR TITLE
JDK-8300042: Improve CPU related JFR events descriptions

### PR DESCRIPTION
**Pull request by MBaesken**
There are a number of JFR events dealing with CPU related information. Those sometimes miss descriptions or have very limited descriptions, this can be improved.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300042](https://bugs.openjdk.org/browse/JDK-8300042): Improve CPU related JFR events descriptions

### Reviewers
 * [Christoph Langer](https://openjdk.org/census#clanger) (@RealCLanger - **Reviewer**)
 * [Lutz Schmidt](https://openjdk.org/census#lucy) (@RealLucy - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/11969/head:pull/11969` \
`$ git checkout pull/11969`

Update a local copy of the PR: \
`$ git checkout pull/11969` \
`$ git pull https://git.openjdk.org/jdk pull/11969/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 11969`

View PR using the GUI difftool: \
`$ git pr show -t 11969`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/11969.diff">https://git.openjdk.org/jdk/pull/11969.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/jfr/metadata/metadata.xml [hunk 1]
```diff
@@ -789,22 +789,28 @@
...
+         period="endChunk">
     <Field type="boolean" name="fastTimeEnabled" label="Fast Time" />
     <Field type="boolean" name="fastTimeAutoEnabled" label="Trusted Platform" />
     <Field type="long" contentType="hertz" name="osFrequency" label="OS Frequency" />
     <Field type="long" contentType="hertz" name="fastTimeFrequency" label="Fast Time Frequency" />
   </Event>
 
-  <Event name="CPULoad" category="Operating System, Processor" label="CPU Load" description="OS CPU Load" period="everyChunk">
+  <Event name="CPULoad" category="Operating System, Processor" label="CPU Load"
+         description="OS CPU Load values of the process, contains information about the recent cpu usage of the JVM process"
```
Comment (RealCLanger) [inline src/hotspot/share/jfr/metadata/metadata.xml:812]: CPU should be written in capitals here, too.

**Review by RealCLanger — CHANGES_REQUESTED** [review]

**Comment by thegreystone** [comment]
The inoffical rule has been to not add descriptions that do not add to the understanding of the event type. Like not adding comments when the code is self explanatory. I would argue that, at least, everything before the comma can be removed.

**Comment by MBaesken** [comment]
Hi Christoph and Marcus, I shortened and adjusted the CPULoad description,

File: src/hotspot/share/jfr/metadata/metadata.xml [hunk 2]
```diff
@@ -789,22 +789,28 @@
     <Field type="string" name="commandLine" label="Command Line" />
   </Event>
 
-  <Event name="CPUInformation" category="Operating System, Processor" label="CPU Information" period="endChunk">
+  <Event name="CPUInformation" category="Operating System, Processor" label="CPU Information"
+         description="Various characteristics and descriptions of the processor(s) the JVM is running on"
```
Comment (RealCLanger) [inline src/hotspot/share/jfr/metadata/metadata.xml:793]: I would leave out the word "Various" here

File: src/hotspot/share/jfr/metadata/metadata.xml [hunk 3]
```diff
@@ -789,22 +789,28 @@
...
     <Field type="string" name="cpu" label="Type" />
     <Field type="string" name="description" label="Description" />
     <Field type="uint" name="sockets" label="Sockets" />
     <Field type="uint" name="cores" label="Cores" />
     <Field type="uint" name="hwThreads" label="Hardware Threads" />
   </Event>
 
-  <Event name="CPUTimeStampCounter" category="Operating System, Processor" label="CPU Time Stamp Counter" period="endChunk">
+  <Event name="CPUTimeStampCounter" category="Operating System, Processor" label="CPU Time Stamp Counter"
+         description="Various information about the CPU time stamp mechanism / (RD)TSC"
```
Comment (RealCLanger) [inline src/hotspot/share/jfr/metadata/metadata.xml:803]: Same here, drop the "Various"

File: src/hotspot/share/jfr/metadata/metadata.xml [hunk 4]
```diff
@@ -789,22 +789,28 @@
...
+         period="endChunk">
     <Field type="boolean" name="fastTimeEnabled" label="Fast Time" />
     <Field type="boolean" name="fastTimeAutoEnabled" label="Trusted Platform" />
     <Field type="long" contentType="hertz" name="osFrequency" label="OS Frequency" />
     <Field type="long" contentType="hertz" name="fastTimeFrequency" label="Fast Time Frequency" />
   </Event>
 
-  <Event name="CPULoad" category="Operating System, Processor" label="CPU Load" description="OS CPU Load" period="everyChunk">
+  <Event name="CPULoad" category="Operating System, Processor" label="CPU Load"
+         description="Contains information about the recent CPU usage of the JVM process"
```
Comment (RealCLanger) [inline src/hotspot/share/jfr/metadata/metadata.xml:812]: Here only say "Information about...", leave out the "Contains"

**Review by RealCLanger — CHANGES_REQUESTED** [review]
I've got 3 further comments to shorten the description texts.

**Comment by MBaesken** [comment]
Hi Christoph, I adjusted the descriptions.

**Review by RealCLanger — APPROVED** [review]
Looks good to me now

**Review by RealLucy — APPROVED** [review]
Looks good to me.

**Comment by MBaesken** [comment]
Hi Lutz and Christoph, thanks for the reviews !

/integrate

**Pull request closed without merging**